services/data/src/components/DataMutation.tsx

```diff
@@ -0,0 +1,24 @@
+import { useDataMutation } from '../hooks/useDataMutation'
+import { Mutation, MutationOptions } from '../engine/types/Mutation'
+import { MutationRenderInput } from '../types'
+
+interface MutationInput extends MutationOptions {
+    mutation: Mutation
+    children: (input: MutationRenderInput) => any
+}
+
+export const DataMutation = ({
+    mutation,
+    onCompleted,
+    onError,
+    variables,
+    children,
+}: MutationInput) => {
+    const mutationState = useDataMutation(mutation, {
+        onCompleted,
```

Comment:
🤷‍♂ I'd usually prefer `on<verb>`, but this came from Apollo Client which uses `onCompleted` for this exact case.  Our react hooks API is almost identical with Apollo's at the moment, which I think is a nice feature.  Do you think it should be `onComplete`?

If we want full parity another consideration is whether to use `client` instead of `engine` - I prefer `engine` there though I think.